addon/utils/compat/get-owner.js

```diff
@@ -0,0 +1,11 @@
+import {
+  macroCondition,
+  dependencySatisfies,
+  importSync,
+} from '@embroider/macros';
+
+export const getOwner = macroCondition(
+  dependencySatisfies('ember-source', '>= 4.11'),
```

Comment:
I would like some confirmation this actually works on older versions, since at least the docs until 5.8 specify importing from `@ember/application`. See for example https://api.emberjs.com/ember/5.8/functions/@ember%2Fowner/getOwner